Mu2eG4/src/constructExtMonFNALBuilding.cc

```diff
@@ -68,169 +96,298 @@ namespace mu2e {
+                                   );
+
+    assert( zPlanes.size() == rInnerSteelShot.size() );
+
+    PolyconsParams steelShotParams ( zPlanes, rInnerSteelShot, rOuterSteelShot );
+
+    VolumeInfo steelShot = nestPolycone(collimator.name()+"SteelShot",
+                                            steelShotParams,
+                                            materialFinder.get("extMonFNAL."+collimator.name()+".shot.materialName"),
+                                            0,
+                                            CLHEP::Hep3Vector(0,0,0),
+                                            collimatorMother,
+                                            0,
+                                            geomOptions->isVisible("extMonFNAL"+collimator.name()+"steelShot"),
+                                            G4Colour::Red(),
+                                            geomOptions->isSolid("extMonFNAL"+collimator.name()+"steelShot"),
+                                            forceAuxEdgeVisible,
+                                            placePV,
+                                            doSurfaceCheck
+                                            );
 
+    //--------------------------------------------------------------------
+    // Inner Shot Liner
 
-    VolumeInfo alignmentHole(collimator.name()+"AlignmentHole",
-                             collimatorCenterInParent,
-                             parent.centerInWorld);
 
-    alignmentHole.solid = new G4IntersectionSolid(alignmentHole.name,
-                                                  parent.solid,
-                                                  holeCylinder,
-                                                  colrot,
-                                                  alignmentHole.centerInParent
-                                                  );
+    vector<double> rInnerInnerShotLinerRadius = {
+      collimator.shotLinerInnerRadius()[1],
+      collimator.shotLinerInnerRadius()[1],
+      collimator.shotLinerInnerRadius()[0],
+      collimator.shotLinerInnerRadius()[0]
+    };
 
-    finishNesting(alignmentHole,
-                  materialFinder.get("hall.insideMaterialName"),
-                  0,
-                  CLHEP::Hep3Vector(0,0,0),
-                  parent.logical,
-                  0,
-                  geomOptions->isVisible("extMonFNAL"+collimator.name()+"alignmentHole"),
-                  G4Colour::Red(),//                  G4Colour::Cyan(),
-                  geomOptions->isSolid("extMonFNAL"+collimator.name()+"alignmentHole"),
-                  forceAuxEdgeVisible,
-                  placePV,
-                  doSurfaceCheck
-                  );
+    vector<double> rOuterInnerShotLinerRadius = rInnerSteelShot;
 
-    //----------------------------------------------------------------
-    // Alignment plug
+    assert( zPlanes.size() == rOuterInnerShotLinerRadius.size() );
```

Comment:
Do you really want an assert here?  Remember that assert is only effective in debug builds.  It is not present in our normal builds, which we all "prof", because it strips the libraries of most debugger symbols, leaving only those necessary for code profiline.  See https://en.cppreference.com/w/cpp/error/assert .  When we do prof builds we define NDEBUG. 

There are other places where you also use assert.

Andrei may want to contribute to this discusision.

Reply:
Hi Rob, I think the assert() use got duplicated from my original code in that file.   I agree that it is better to change it to throwing a cet::exception.

Reply:
I suggest that we make a global change of this a separate PR.   Maybe encapsulate it in a small function that takes __FILE__ and __LINE__ as arguments?  Does that work for you?